pkgs/development/python-modules/cynthion/default.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  pythonOlder,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  amaranth,
+  apollo-fpga,
+  future,
+  libusb1,
+  luna-soc,
+  luna-usb,
+  prompt-toolkit,
+  pyfwup,
+  pygreat,
+  pyserial,
+  pyusb,
+  tabulate,
+  tomli,
+  tqdm,
+
+  # tests
+  pytestCheckHook,
+}:
+buildPythonPackage rec {
+  pname = "cynthion";
+  version = "0.1.7";
+  pyproject = true;
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "greatscottgadgets";
+    repo = "cynthion";
+    rev = "refs/tags/${version}";
+    hash = "sha256-2nVfODAg4t5hoSKUEP4IN23R+JGe3lw/rpfjW/UIsYw=";
+  };
+
+  sourceRoot = "${src.name}/cynthion/python";
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail '"setuptools-git-versioning<2"' "" \
+      --replace-fail 'dynamic = ["version"]' 'version = "${version}"'
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    amaranth
+    apollo-fpga
+    future
+    libusb1
+    luna-soc
+    luna-usb
+    prompt-toolkit
+    pyfwup
+    pygreat
+    pyserial
+    pyusb
+    tabulate
+    tomli
+    tqdm
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "cynthion" ];
+
+  meta = {
+    changelog = "https://github.com/greatscottgadgets/cynthion/releases/tag/${version}";
+    description = "Python package and utilities for the Great Scott Gadgets Cynthion USB Test Instrument";
+    homepage = "https://github.com/greatscottgadgets/cynthion";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ carlossless ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

Comment:
`buildPythonPackage` sets `meta.platforms = lib.platforms.unix` by default.
Is there any reason to limit it to linux and darwin only?

Reply:
Actually no. Thanks for spotting this!

I think I added this because I thought those two were the officially supported platforms, but that's actually not the case. Supported platforms are not mentioned.

I removed it now.